packages/loader/container-loader/src/retriableDocumentStorageService.ts

```diff
@@ -89,17 +109,40 @@ export class RetriableDocumentStorageService implements IDocumentStorageService
                 if (!canRetryOnError(err)) {
                     throw err;
                 }
+                numRetries += 1;
+                lastError = err;
                 // If the error is throttling error, then wait for the specified time before retrying.
                 // If the waitTime is not specified, then we start with retrying immediately to max of 8s.
                 retryAfter = getRetryDelayFromError(err) ?? Math.min(retryAfter * 2, 8000);
+                totalWaitTime += retryAfter;
                 if (id === undefined) {
                     id = uuid();
                 }
                 this.deltaManager.emitDelayInfo(id, retryAfter, CreateContainerError(err));
                 await this.delay(retryAfter);
             }
         } while (!success);
+        if (numRetries > 0) {
+            this.logger.sendTelemetryEvent({
+                eventName: "StorageCallsRetried",
```

Comment:
In my experience, it's easier to consume telemetry like that if eventName = `StorageCallsRetried_${fetchCallName}` and not use extra field for that. Basically we have too many fields in Kusto and it's really hard (without look at the code) what fields you need, so making sure that 95% of what you need can be accomplished by looking at eventName, error & reason fields helps OCEs be productive.